javascript/revision_delete.js

```diff
@@ -0,0 +1,89 @@
+/*
+    revision_delete.js
+
+    MediaWiki API Demos
+    Demo of `Revisiondelete` module: Hide all information about revision ID 71. 
+    (The target, Sample Page, is unnecessary in this case.)
+
+    MIT license
+*/
+var request = require("request").defaults({jar: true}),
+url = "https://test.wikipedia.org/w/api.php";
+
+// Step 1: GET Request to fetch login token
+function getLoginToken() {
+    var params_0 = {
+        action: "query",
+        meta: "tokens",
+        type: "login",
+        format: "json"
+    };
+    request.get({ url: url, qs: params_0 }, function (error, res, body) {
+        if (error) {
+            return;
+        }
+        var data = JSON.parse(body);
+        loginRequest(data.query.tokens.logintoken);
+    });
+}
+
+// Step 2: Send a post request to log in using the clientlogin method.
```

Comment:
This comment mentions `import` rights but this module does not use import rights. Also I'm wondering why we are using the clientlogin method to login yet `deleterevision` rights can be granted through the `Special:Botpasswords` page?